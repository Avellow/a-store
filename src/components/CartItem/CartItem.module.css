.item {
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 20px;
}

.info {
  display: flex;
  justify-content: space-between;
  gap: 15px;
  align-items: center;
  flex: 1;
}

.title {
  font-style: italic;
  margin-bottom: 5px;
}

.options {
  list-style: none;
  padding: 0;
  margin: 0;
  display: flex;
  flex-direction: column;
  gap: 2px;
}

.option {
  opacity: 0.7;
  font-family: var(--my-font-family);
}

.options li {
  padding: 0;
}

.control {
  display: flex;
  align-items: center;
}

.control span {
  font-weight: bold;
  padding: 0 2px;
}

.option {
  font-size: 12px;
  line-height: 1.1;
  margin: 0;
}

.quantityControl {
  font-family: var(--my-font-family);
  display: flex;
  justify-content: space-between;
  align-items: center;
  gap: 10px;
  min-width: 185px;
}

.controlIcon {
  opacity: 0.5;
}

.price {
  font-family: var(--my-font-family);
}

@media screen and (max-width: 675px) {
  .item {
    gap: 10px;
  }
}

@media screen and (max-width: 480px) {
  .info {
    flex-direction: column;
    align-items: baseline;
  }

  .quantityControl {
    flex-wrap: wrap;
    gap: 3px;
    min-width: 0;
  }
}
