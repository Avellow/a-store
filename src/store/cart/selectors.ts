import { ApplicationState } from "..";

export const cartStateSelector = (state: ApplicationState) => state.cart;
export const cartItemsSelector = (state: ApplicationState) =>
  cartStateSelector(state).items;
export const hasErrorSelector = (state: ApplicationState) =>
  cartStateSelector(state).hasError;
export const isCartHasItemsSelector = (state: ApplicationState) =>
  !!cartStateSelector(state).items.length;

export const itemsQuantitySelector = (state: ApplicationState) =>
  cartStateSelector(state).items.reduce(
    (acc, { quantity }) => acc + quantity,
    0
  );

export const totalAmountSelector = (state: ApplicationState) =>
  cartStateSelector(state).items.reduce(
    (acc, { quantity, price }) => acc + price * quantity,
    0
  );
