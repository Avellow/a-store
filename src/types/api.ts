type size = "XXS" | "XS" | "S" | "M" | "L" | "XL" | "XXL";

export type ProductType = {
  id: number;
  preview: string;
  images?: string[];
  title: string;
  subtitle?: string;
  price: number;
  description?: string;
  colors?: string[];
  sizes?: size[];
  models?: string[];
  stickers?: string[];
  stickerNumbers?: number[];
  availability: boolean;
};

type ProductsGroup = {
  id: number;
  title: string;
  description: string;
  products: ProductType[];
};

type CartItemOptionsType = Record<string, string | number | undefined>;

type CartItemType = {
  title: string;
  imageURL: string;
  options?: CartItemOptionsType;
  quantity: number;
  price: number;
};

export {
  ProductsGroup,
  ProductType as Product,
  CartItemType,
  CartItemOptionsType,
};
