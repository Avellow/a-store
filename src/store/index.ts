import createSagaMiddleware from "@redux-saga/core";
import { combineReducers, configureStore } from "@reduxjs/toolkit";
import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
import { productsReducer } from "./alfa-products";
import { cartReducer } from "./cart";
import { groupsReducer } from "./design-groups";
import { notificationsReducer } from "./notifications";
import { rootSaga } from "./root-saga";

const sagaMiddleware = createSagaMiddleware();

export const appReducer = combineReducers({
  products: productsReducer,
  groups: groupsReducer,
  notifications: notificationsReducer,
  cart: cartReducer,
});

export const store = configureStore({
  reducer: appReducer,
  devTools: true,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(sagaMiddleware),
});

sagaMiddleware.run(rootSaga);

export type ApplicationState = ReturnType<typeof store.getState>;
export const useAppDispatch = () => useDispatch<typeof store.dispatch>();
export const useAppSelector: TypedUseSelectorHook<ApplicationState> =
  useSelector;
