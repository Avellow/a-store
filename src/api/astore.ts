import axios from "axios";
import { ProductType, ProductsGroup } from "../types/api";

const astoreApiUrl = "http://qa-games.ru/astore";

export const getYourDesignGroups = async (): Promise<ProductsGroup[]> => {
  const response = await axios.get<ProductsGroup[]>(
    `${astoreApiUrl}/your-design`
  );
  return response.data;
};

export const getMadeInAlfaProducts = async (): Promise<ProductType[]> => {
  const response = await axios.get<ProductType[]>(
    `${astoreApiUrl}/made-in-alfa`
  );
  return response.data;
};

export const getProduct = async (
  id: number
): Promise<ProductType | undefined> => {
  const response = await axios.get<ProductType>(
    `${astoreApiUrl}/product/${id}`
  );
  return response.data;
};

export const createOrder = async () => {
  /* заглушка */
};
