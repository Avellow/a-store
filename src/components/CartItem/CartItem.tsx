import { useMemo } from 'react';
import { SuperEllipse } from '@alfalab/core-components/icon-view/super-ellipse';
import { Typography } from '@alfalab/core-components/typography';
import { IconButton } from '@alfalab/core-components/icon-button';
import { AddMIcon } from '@alfalab/icons-glyph/AddMIcon';
import { MinusMIcon } from '@alfalab/icons-glyph/MinusMIcon';
import { Amount } from '@alfalab/core-components/amount';
import { DeleteMBlackIcon } from '@alfalab/icons-classic/DeleteMBlackIcon';

import styles from './CartItem.module.css';
import { CartItemProps } from './CartItem.props';
import { translateToRu } from '../../vendor/engToRuDictionary';
import { useAppDispatch } from '../../store';
import { cartActions } from '../../store/cart';

export const CartItem = ({ item }: CartItemProps): JSX.Element => {

  const dispatch = useAppDispatch();
  const { options, imageURL, title, quantity, price } = item;

  const optionsEntries = useMemo(() => {
    if (options) {
      return Object.entries(options);
    }
  }, [options]);

  const handleDecrease = () => {
    if (item.quantity === 1) {
      dispatch(cartActions.removeItem(item));
    } else {
      dispatch(cartActions.decreaseQuantity(item));
    }
  };

  const handleIncrease = () => {
    dispatch(cartActions.addItem(item));
  };

  const handleRemove = () => {
    dispatch(cartActions.removeItem(item))
  };

  return (
    <div className={styles.item}>
      <SuperEllipse
        size={80}
        imageUrl={imageURL}
        dataTestId='img'
      />
      <div className={styles.info}>
        <div className={styles.details}>
          <Typography.TitleResponsive
            font='styrene'
            view='xsmall'
            weight='bold'
            color='primary'
            tag='h4'
            className={styles.title}
            dataTestId='title'
          >
            {title}
          </Typography.TitleResponsive>

          {!!optionsEntries?.length && (
            <ul className={styles.options}>
              {optionsEntries.map(([name, value]) => (
                <li key={name}>
                  <Typography.Text
                    view='primary-small'
                    weight='bold'
                    className={styles.option}
                    dataTestId={`${name}-option`}
                  >
                    {translateToRu(name)}: {translateToRu(`${value}`)}
                  </Typography.Text>
                </li>
              ))}
            </ul>
          )}

        </div>
        <div className={styles.quantityControl}>
          <div className={styles.control}>
            <IconButton
              icon={MinusMIcon}
              size='xs'
              className={styles.controlIcon}
              onClick={handleDecrease}
              dataTestId='decrease-quantity'
            />
            <span data-test-id='quantity'>{quantity}</span>
            <IconButton
              icon={AddMIcon}
              size='xs'
              className={styles.controlIcon}
              onClick={handleIncrease}
              dataTestId='increase-quantity'
            />
          </div>
          <Amount
            value={quantity * price}
            minority={0}
            currency='RUB'
            className={styles.price}
            dataTestId='total-price'
          />
        </div>
      </div>
      <IconButton icon={DeleteMBlackIcon} colors='inverted' onClick={handleRemove} dataTestId='remove-item' />
    </div>
  );
};
