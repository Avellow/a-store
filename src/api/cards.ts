import axios from "axios";
import { ProductType, ProductsGroup } from "./../types/api";

const apiUrl = "/products.json";
const groupsUrl = "/groups.json";

export const getProducts = (): Promise<ProductType[]> => {
  return axios.get(apiUrl).then((response) => response.data.products);
};

// временное решение дял заглушки без обработки ошибок
export const getProduct = async (
  id: number
): Promise<ProductType | undefined> => {
  const products = await getProducts();
  return products.find((product) => product.id === id);
};

export const getGroups = (): Promise<ProductsGroup[]> => {
  return axios.get(groupsUrl).then((response) => response.data.groups);
};
