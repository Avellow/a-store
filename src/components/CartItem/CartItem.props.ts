import { CartItemType } from "../../types/api";

export interface CartItemProps {
  item: CartItemType;
}
