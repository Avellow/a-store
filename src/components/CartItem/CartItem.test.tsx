import { configure, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import * as reduxHooks from '../../store';
import { cartActions } from '../../store/cart';
import { cartTestingItem1, cartTestingItem2 } from '../../tests/helpers/cartItems';
import { CartItem } from './CartItem';
import { renderWithRouterAndProvider } from '../../tests/helpers/renderWithRouterAndProvider';
import { CartItemType } from '../../types/api';

configure({ testIdAttribute: 'data-test-id' });

const renderComponent = (item: CartItemType) =>
  renderWithRouterAndProvider(<CartItem item={item} />);

describe('Cart-item component view', () => {

  it('should be rendered', () => {
    renderComponent(cartTestingItem1);
    expect(renderComponent(cartTestingItem2)).toMatchSnapshot();
  });

  it('should render product-description correctly', () => {
    renderComponent(cartTestingItem1);
    expect(screen.getByTestId('img')).toBeInTheDocument();
    expect(screen.getByTestId('title')).toBeInTheDocument();
    expect(screen.getByTestId('color-option')).toBeInTheDocument();
    expect(screen.getByTestId('size-option')).toBeInTheDocument();
    expect(screen.getByTestId('stickerNumber-option')).toBeInTheDocument();
    expect(screen.queryByTestId('model-option')).not.toBeInTheDocument();
    expect(screen.getByTestId('quantity')).toBeInTheDocument();
    expect(screen.getByTestId('quantity')).toHaveTextContent(`${cartTestingItem1.quantity}`);
    expect(screen.getByTestId('total-price')).toHaveTextContent('3 598');
  });
});

describe('Cart-item actions', () => {

  const mockedUseDispatch = jest.spyOn(reduxHooks, 'useAppDispatch');

  beforeEach(() => {
    mockedUseDispatch.mockReturnValue(jest.fn());
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  it('should dispatch increase action', () => {
    const dispatch = jest.fn();
    mockedUseDispatch.mockReturnValue(dispatch);

    const mockedIncreaseQuantity = jest.spyOn(cartActions, 'addItem');

    renderComponent(cartTestingItem1);

    const increaseButton = screen.getByTestId('increase-quantity');

    fireEvent.click(increaseButton);

    expect(dispatch).toHaveBeenCalledTimes(1);
    expect(mockedIncreaseQuantity).toHaveBeenCalled();
  });

  it('should dispatch decrease action', () => {
    const dispatch = jest.fn();
    mockedUseDispatch.mockReturnValue(dispatch);

    const mockedDecreaseQuantity = jest.spyOn(cartActions, 'decreaseQuantity');

    renderComponent(cartTestingItem2);

    const decreaseButton = screen.getByTestId('decrease-quantity');

    fireEvent.click(decreaseButton);

    expect(dispatch).toHaveBeenCalledTimes(1);
    expect(mockedDecreaseQuantity).toHaveBeenCalled();
  });

  it('should dispatch remove action', () => {
    const dispatch = jest.fn();
    mockedUseDispatch.mockReturnValue(dispatch);

    const mockedRemoveItem = jest.spyOn(cartActions, 'removeItem');

    renderComponent(cartTestingItem2);

    const removeButton = screen.getByTestId('remove-item');

    fireEvent.click(removeButton);

    expect(dispatch).toHaveBeenCalledTimes(1);
    expect(mockedRemoveItem).toHaveBeenCalled();
  });
});
