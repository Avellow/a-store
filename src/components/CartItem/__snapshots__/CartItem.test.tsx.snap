// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart-item component view should be rendered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="item"
      >
        <div
          class="icon-view__componentWrapper_1wtym icon-view__size_80_1wtym"
          data-test-id="img"
        >
          <div
            class="icon-view__component_1wtym"
          >
            <svg
              focusable="false"
              height="80"
              viewBox="0 0 80 80"
              width="80"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="icon-view__bg_1wtym"
                d="M1.30048 20.5263C2.75571 10.3588 10.3588 2.75571 20.5263 1.30048C25.7164 0.557648 32.2076 0 40 0C47.7924 0 54.2836 0.557648 59.4737 1.30048C69.6412 2.75571 77.2443 10.3588 78.6995 20.5263C79.4423 25.7164 80 32.2076 80 40C80 47.7924 79.4423 54.2836 78.6995 59.4737C77.2443 69.6412 69.6412 77.2443 59.4737 78.6995C54.2836 79.4423 47.7924 80 40 80C32.2076 80 25.7164 79.4423 20.5263 78.6995C10.3588 77.2443 2.75571 69.6412 1.30048 59.4737C0.557648 54.2836 0 47.7924 0 40C0 32.2076 0.557648 25.7164 1.30048 20.5263Z"
                style="fill: var(--color-light-bg-secondary);"
              />
              <defs>
                <pattern
                  height="100%"
                  id="https68519498jpg_80"
                  width="100%"
                >
                  <image
                    height="100%"
                    href="https://68519498.jpg"
                    preserveAspectRatio="xMidYMid slice"
                    width="100%"
                  />
                </pattern>
              </defs>
              <path
                d="M1.30048 20.5263C2.75571 10.3588 10.3588 2.75571 20.5263 1.30048C25.7164 0.557648 32.2076 0 40 0C47.7924 0 54.2836 0.557648 59.4737 1.30048C69.6412 2.75571 77.2443 10.3588 78.6995 20.5263C79.4423 25.7164 80 32.2076 80 40C80 47.7924 79.4423 54.2836 78.6995 59.4737C77.2443 69.6412 69.6412 77.2443 59.4737 78.6995C54.2836 79.4423 47.7924 80 40 80C32.2076 80 25.7164 79.4423 20.5263 78.6995C10.3588 77.2443 2.75571 69.6412 1.30048 59.4737C0.557648 54.2836 0 47.7924 0 40C0 32.2076 0.557648 25.7164 1.30048 20.5263Z"
                style="fill: url(#https68519498jpg_80);"
              />
            </svg>
          </div>
        </div>
        <div
          class="info"
        >
          <div
            class="details"
          >
            <h4
              class="typography__component_h1nzy title typography__styrene-xsmall_qohs3 typography__bold_h1nzy typography__primary_1faea"
              data-test-id="title"
            >
              Тестовое худи
            </h4>
            <ul
              class="options"
            >
              <li>
                <span
                  class="option typography__primary-small_xcdcz typography__bold_xcdcz"
                  data-test-id="stickerNumber-option"
                >
                  номер стикера
                  : 
                  1
                </span>
              </li>
              <li>
                <span
                  class="option typography__primary-small_xcdcz typography__bold_xcdcz"
                  data-test-id="color-option"
                >
                  цвет
                  : 
                  белый
                </span>
              </li>
              <li>
                <span
                  class="option typography__primary-small_xcdcz typography__bold_xcdcz"
                  data-test-id="size-option"
                >
                  размер
                  : 
                  M
                </span>
              </li>
            </ul>
          </div>
          <div
            class="quantityControl"
          >
            <div
              class="control"
            >
              <button
                class="button__component_onnkc button__ghost_onnkc button__s_onnkc button__component_1qjkj button__ghost_1qjkj cc-icon-button controlIcon icon-button__primary_7jncp icon-button__component_7jncp"
                data-test-id="decrease-quantity"
                type="button"
              >
                <span
                  class="button__text_onnkc button__stretchText_onnkc"
                >
                  <span
                    class="icon-button__iconWrapper_gy47p icon-button__xs_gy47p"
                  >
                    <svg
                      class="icon-button__icon_gy47p"
                      fill="currentColor"
                      focusable="false"
                      height="24"
                      role="img"
                      viewBox="0 0 24 24"
                      width="24"
                    >
                      <path
                        d="M3 11h18v2H3z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <span
                data-test-id="quantity"
              >
                2
              </span>
              <button
                class="button__component_onnkc button__ghost_onnkc button__s_onnkc button__component_1qjkj button__ghost_1qjkj cc-icon-button controlIcon icon-button__primary_7jncp icon-button__component_7jncp"
                data-test-id="increase-quantity"
                type="button"
              >
                <span
                  class="button__text_onnkc button__stretchText_onnkc"
                >
                  <span
                    class="icon-button__iconWrapper_gy47p icon-button__xs_gy47p"
                  >
                    <svg
                      class="icon-button__icon_gy47p"
                      fill="currentColor"
                      focusable="false"
                      height="24"
                      role="img"
                      viewBox="0 0 24 24"
                      width="24"
                    >
                      <path
                        d="M11 11H5v2h6v6h2v-6h6v-2h-6V5h-2v6z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
            </div>
            <span
              class="amount__component_96zmx price amount__defaultStyles_96zmx"
              data-test-id="total-price"
            >
              3 598
              <span
                class="amount__defaultMinor_96zmx"
              >
                 ₽
              </span>
            </span>
          </div>
        </div>
        <button
          class="button__component_onnkc button__ghost_onnkc button__s_onnkc button__component_1qjkj button__ghost_1qjkj cc-icon-button icon-button__primary_jaon2 icon-button__component_jaon2"
          data-test-id="remove-item"
          type="button"
        >
          <span
            class="button__text_onnkc button__stretchText_onnkc"
          >
            <span
              class="icon-button__iconWrapper_gy47p icon-button__s_gy47p"
            >
              <svg
                class="icon-button__icon_gy47p"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                >
                  <path
                    d="M0 0h24v24H0z"
                  />
                  <path
                    d="M4 8v12.5C4 21.878 5.122 23 6.5 23h12c1.378 0 2.5-1.122 2.5-2.5V8h-1v12.5c0 .827-.673 1.5-1.5 1.5h-12c-.827 0-1.5-.673-1.5-1.5V8H4zM3 5.5C3 4.673 3.673 4 4.5 4h16c.827 0 1.5.673 1.5 1.5V8h1V5.5C23 4.122 21.878 3 20.5 3H16V1.5c0-.827-.673-1.5-1.5-1.5h-4C9.673 0 9 .673 9 1.5V3H4.5A2.503 2.503 0 002 5.5V8h1V5.5zm7-4a.5.5 0 01.5-.5h4a.5.5 0 01.5.5V3h-5V1.5z"
                    fill="#0B1F35"
                  />
                  <path
                    d="M9 9h2v9H9V9zm5 0h2v9h-2V9z"
                    fill="#0B1F35"
                    opacity=".3"
                  />
                </g>
              </svg>
            </span>
          </span>
        </button>
      </div>
    </div>
    <div>
      <div
        class="item"
      >
        <div
          class="icon-view__componentWrapper_1wtym icon-view__size_80_1wtym"
          data-test-id="img"
        >
          <div
            class="icon-view__component_1wtym"
          >
            <svg
              focusable="false"
              height="80"
              viewBox="0 0 80 80"
              width="80"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="icon-view__bg_1wtym"
                d="M1.30048 20.5263C2.75571 10.3588 10.3588 2.75571 20.5263 1.30048C25.7164 0.557648 32.2076 0 40 0C47.7924 0 54.2836 0.557648 59.4737 1.30048C69.6412 2.75571 77.2443 10.3588 78.6995 20.5263C79.4423 25.7164 80 32.2076 80 40C80 47.7924 79.4423 54.2836 78.6995 59.4737C77.2443 69.6412 69.6412 77.2443 59.4737 78.6995C54.2836 79.4423 47.7924 80 40 80C32.2076 80 25.7164 79.4423 20.5263 78.6995C10.3588 77.2443 2.75571 69.6412 1.30048 59.4737C0.557648 54.2836 0 47.7924 0 40C0 32.2076 0.557648 25.7164 1.30048 20.5263Z"
                style="fill: var(--color-light-bg-secondary);"
              />
              <defs>
                <pattern
                  height="100%"
                  id="https68519498jpg_80"
                  width="100%"
                >
                  <image
                    height="100%"
                    href="https://68519498.jpg"
                    preserveAspectRatio="xMidYMid slice"
                    width="100%"
                  />
                </pattern>
              </defs>
              <path
                d="M1.30048 20.5263C2.75571 10.3588 10.3588 2.75571 20.5263 1.30048C25.7164 0.557648 32.2076 0 40 0C47.7924 0 54.2836 0.557648 59.4737 1.30048C69.6412 2.75571 77.2443 10.3588 78.6995 20.5263C79.4423 25.7164 80 32.2076 80 40C80 47.7924 79.4423 54.2836 78.6995 59.4737C77.2443 69.6412 69.6412 77.2443 59.4737 78.6995C54.2836 79.4423 47.7924 80 40 80C32.2076 80 25.7164 79.4423 20.5263 78.6995C10.3588 77.2443 2.75571 69.6412 1.30048 59.4737C0.557648 54.2836 0 47.7924 0 40C0 32.2076 0.557648 25.7164 1.30048 20.5263Z"
                style="fill: url(#https68519498jpg_80);"
              />
            </svg>
          </div>
        </div>
        <div
          class="info"
        >
          <div
            class="details"
          >
            <h4
              class="typography__component_h1nzy title typography__styrene-xsmall_qohs3 typography__bold_h1nzy typography__primary_1faea"
              data-test-id="title"
            >
              Тестовый рюкзак
            </h4>
            <ul
              class="options"
            />
          </div>
          <div
            class="quantityControl"
          >
            <div
              class="control"
            >
              <button
                class="button__component_onnkc button__ghost_onnkc button__s_onnkc button__component_1qjkj button__ghost_1qjkj cc-icon-button controlIcon icon-button__primary_7jncp icon-button__component_7jncp"
                data-test-id="decrease-quantity"
                type="button"
              >
                <span
                  class="button__text_onnkc button__stretchText_onnkc"
                >
                  <span
                    class="icon-button__iconWrapper_gy47p icon-button__xs_gy47p"
                  >
                    <svg
                      class="icon-button__icon_gy47p"
                      fill="currentColor"
                      focusable="false"
                      height="24"
                      role="img"
                      viewBox="0 0 24 24"
                      width="24"
                    >
                      <path
                        d="M3 11h18v2H3z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
              <span
                data-test-id="quantity"
              >
                3
              </span>
              <button
                class="button__component_onnkc button__ghost_onnkc button__s_onnkc button__component_1qjkj button__ghost_1qjkj cc-icon-button controlIcon icon-button__primary_7jncp icon-button__component_7jncp"
                data-test-id="increase-quantity"
                type="button"
              >
                <span
                  class="button__text_onnkc button__stretchText_onnkc"
                >
                  <span
                    class="icon-button__iconWrapper_gy47p icon-button__xs_gy47p"
                  >
                    <svg
                      class="icon-button__icon_gy47p"
                      fill="currentColor"
                      focusable="false"
                      height="24"
                      role="img"
                      viewBox="0 0 24 24"
                      width="24"
                    >
                      <path
                        d="M11 11H5v2h6v6h2v-6h6v-2h-6V5h-2v6z"
                      />
                    </svg>
                  </span>
                </span>
              </button>
            </div>
            <span
              class="amount__component_96zmx price amount__defaultStyles_96zmx"
              data-test-id="total-price"
            >
              11 997
              <span
                class="amount__defaultMinor_96zmx"
              >
                 ₽
              </span>
            </span>
          </div>
        </div>
        <button
          class="button__component_onnkc button__ghost_onnkc button__s_onnkc button__component_1qjkj button__ghost_1qjkj cc-icon-button icon-button__primary_jaon2 icon-button__component_jaon2"
          data-test-id="remove-item"
          type="button"
        >
          <span
            class="button__text_onnkc button__stretchText_onnkc"
          >
            <span
              class="icon-button__iconWrapper_gy47p icon-button__s_gy47p"
            >
              <svg
                class="icon-button__icon_gy47p"
                focusable="false"
                height="24"
                role="img"
                viewBox="0 0 24 24"
                width="24"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                >
                  <path
                    d="M0 0h24v24H0z"
                  />
                  <path
                    d="M4 8v12.5C4 21.878 5.122 23 6.5 23h12c1.378 0 2.5-1.122 2.5-2.5V8h-1v12.5c0 .827-.673 1.5-1.5 1.5h-12c-.827 0-1.5-.673-1.5-1.5V8H4zM3 5.5C3 4.673 3.673 4 4.5 4h16c.827 0 1.5.673 1.5 1.5V8h1V5.5C23 4.122 21.878 3 20.5 3H16V1.5c0-.827-.673-1.5-1.5-1.5h-4C9.673 0 9 .673 9 1.5V3H4.5A2.503 2.503 0 002 5.5V8h1V5.5zm7-4a.5.5 0 01.5-.5h4a.5.5 0 01.5.5V3h-5V1.5z"
                    fill="#0B1F35"
                  />
                  <path
                    d="M9 9h2v9H9V9zm5 0h2v9h-2V9z"
                    fill="#0B1F35"
                    opacity=".3"
                  />
                </g>
              </svg>
            </span>
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="item"
    >
      <div
        class="icon-view__componentWrapper_1wtym icon-view__size_80_1wtym"
        data-test-id="img"
      >
        <div
          class="icon-view__component_1wtym"
        >
          <svg
            focusable="false"
            height="80"
            viewBox="0 0 80 80"
            width="80"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              class="icon-view__bg_1wtym"
              d="M1.30048 20.5263C2.75571 10.3588 10.3588 2.75571 20.5263 1.30048C25.7164 0.557648 32.2076 0 40 0C47.7924 0 54.2836 0.557648 59.4737 1.30048C69.6412 2.75571 77.2443 10.3588 78.6995 20.5263C79.4423 25.7164 80 32.2076 80 40C80 47.7924 79.4423 54.2836 78.6995 59.4737C77.2443 69.6412 69.6412 77.2443 59.4737 78.6995C54.2836 79.4423 47.7924 80 40 80C32.2076 80 25.7164 79.4423 20.5263 78.6995C10.3588 77.2443 2.75571 69.6412 1.30048 59.4737C0.557648 54.2836 0 47.7924 0 40C0 32.2076 0.557648 25.7164 1.30048 20.5263Z"
              style="fill: var(--color-light-bg-secondary);"
            />
            <defs>
              <pattern
                height="100%"
                id="https68519498jpg_80"
                width="100%"
              >
                <image
                  height="100%"
                  href="https://68519498.jpg"
                  preserveAspectRatio="xMidYMid slice"
                  width="100%"
                />
              </pattern>
            </defs>
            <path
              d="M1.30048 20.5263C2.75571 10.3588 10.3588 2.75571 20.5263 1.30048C25.7164 0.557648 32.2076 0 40 0C47.7924 0 54.2836 0.557648 59.4737 1.30048C69.6412 2.75571 77.2443 10.3588 78.6995 20.5263C79.4423 25.7164 80 32.2076 80 40C80 47.7924 79.4423 54.2836 78.6995 59.4737C77.2443 69.6412 69.6412 77.2443 59.4737 78.6995C54.2836 79.4423 47.7924 80 40 80C32.2076 80 25.7164 79.4423 20.5263 78.6995C10.3588 77.2443 2.75571 69.6412 1.30048 59.4737C0.557648 54.2836 0 47.7924 0 40C0 32.2076 0.557648 25.7164 1.30048 20.5263Z"
              style="fill: url(#https68519498jpg_80);"
            />
          </svg>
        </div>
      </div>
      <div
        class="info"
      >
        <div
          class="details"
        >
          <h4
            class="typography__component_h1nzy title typography__styrene-xsmall_qohs3 typography__bold_h1nzy typography__primary_1faea"
            data-test-id="title"
          >
            Тестовый рюкзак
          </h4>
          <ul
            class="options"
          />
        </div>
        <div
          class="quantityControl"
        >
          <div
            class="control"
          >
            <button
              class="button__component_onnkc button__ghost_onnkc button__s_onnkc button__component_1qjkj button__ghost_1qjkj cc-icon-button controlIcon icon-button__primary_7jncp icon-button__component_7jncp"
              data-test-id="decrease-quantity"
              type="button"
            >
              <span
                class="button__text_onnkc button__stretchText_onnkc"
              >
                <span
                  class="icon-button__iconWrapper_gy47p icon-button__xs_gy47p"
                >
                  <svg
                    class="icon-button__icon_gy47p"
                    fill="currentColor"
                    focusable="false"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <path
                      d="M3 11h18v2H3z"
                    />
                  </svg>
                </span>
              </span>
            </button>
            <span
              data-test-id="quantity"
            >
              3
            </span>
            <button
              class="button__component_onnkc button__ghost_onnkc button__s_onnkc button__component_1qjkj button__ghost_1qjkj cc-icon-button controlIcon icon-button__primary_7jncp icon-button__component_7jncp"
              data-test-id="increase-quantity"
              type="button"
            >
              <span
                class="button__text_onnkc button__stretchText_onnkc"
              >
                <span
                  class="icon-button__iconWrapper_gy47p icon-button__xs_gy47p"
                >
                  <svg
                    class="icon-button__icon_gy47p"
                    fill="currentColor"
                    focusable="false"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <path
                      d="M11 11H5v2h6v6h2v-6h6v-2h-6V5h-2v6z"
                    />
                  </svg>
                </span>
              </span>
            </button>
          </div>
          <span
            class="amount__component_96zmx price amount__defaultStyles_96zmx"
            data-test-id="total-price"
          >
            11 997
            <span
              class="amount__defaultMinor_96zmx"
            >
               ₽
            </span>
          </span>
        </div>
      </div>
      <button
        class="button__component_onnkc button__ghost_onnkc button__s_onnkc button__component_1qjkj button__ghost_1qjkj cc-icon-button icon-button__primary_jaon2 icon-button__component_jaon2"
        data-test-id="remove-item"
        type="button"
      >
        <span
          class="button__text_onnkc button__stretchText_onnkc"
        >
          <span
            class="icon-button__iconWrapper_gy47p icon-button__s_gy47p"
          >
            <svg
              class="icon-button__icon_gy47p"
              focusable="false"
              height="24"
              role="img"
              viewBox="0 0 24 24"
              width="24"
            >
              <g
                fill="none"
                fill-rule="evenodd"
              >
                <path
                  d="M0 0h24v24H0z"
                />
                <path
                  d="M4 8v12.5C4 21.878 5.122 23 6.5 23h12c1.378 0 2.5-1.122 2.5-2.5V8h-1v12.5c0 .827-.673 1.5-1.5 1.5h-12c-.827 0-1.5-.673-1.5-1.5V8H4zM3 5.5C3 4.673 3.673 4 4.5 4h16c.827 0 1.5.673 1.5 1.5V8h1V5.5C23 4.122 21.878 3 20.5 3H16V1.5c0-.827-.673-1.5-1.5-1.5h-4C9.673 0 9 .673 9 1.5V3H4.5A2.503 2.503 0 002 5.5V8h1V5.5zm7-4a.5.5 0 01.5-.5h4a.5.5 0 01.5.5V3h-5V1.5z"
                  fill="#0B1F35"
                />
                <path
                  d="M9 9h2v9H9V9zm5 0h2v9h-2V9z"
                  fill="#0B1F35"
                  opacity=".3"
                />
              </g>
            </svg>
          </span>
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
